packages/teams-js/src/private/externalAppCommands.ts

```diff
@@ -0,0 +1,192 @@
+
+/**
+ * @hidden
+ * Namespace to delegate the ActionCommand to the host
+ * @internal
+ * Limited to Microsoft-internal use
+ *
+ * @beta
+ */
+export namespace externalAppCommands {
+  /**
+   * @hidden
+   * The payload of IActionCommandResponse
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface IActionCommandResponse {
+    /**
+     * type will be result or message
+     */
+    type?: ResultTypes | null;
+    /**
+     * attachment items will exist only when type is result
+     */
+    items?: MessageExtensionItem[] | null;
+    /**
+     * text will exist only when type is message
+     */
+    text?: string | null;
+  }
+
+  /**
+   * @hidden
+   * The result type for the ActionCommandResponse
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export type ResultTypes = 'result' | 'message';
+
+  /**
+   * @hidden
+   * The payload of attachment item
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface MessageExtensionItem {
+    card: Card;
+    previewCard?: Card | null;
+  }
+
+  /**
+   * @hidden
+   * The payload of Card
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface Card {
+    appIcon?: string | null;
+    appId?: string | null;
+    appName?: string | null;
+    cardClientId?: string | null;
+    content: string;
+    contentType: string;
+    locale?: string | null;
+    name?: string | null;
+    userSpecificContent?: UserSpecificCardContent | null;
+    sourceType?: CardSourceType | null;
+  }
+
+  /**
+   * @hidden
+   * The payload that is used when executing an Adaptive Card Action.Submit
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface UserSpecificCardContent {
+    content?: string | null;
+    timeStamp: number;
+    version?: string | null;
+    authBlockLastUsedOn?: number | null;
+  }
+
+  /**
+   * @hidden
+   * The payload that is used for CardSourceType
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export type CardSourceType =
+    | 'ZeroInstallAppInstalledAppLinkUnfurl'
+    | 'NonZeroInstallAppInstalledAppLinkUnfurl'
+    | 'ZeroInstallAppNotInstalledAppLinkUnfurl'
+    | 'URLP'
+    | 'URLPRichInfo'
+    | 'NonLinkUnfurling';
+
+  /**
+   *
+   * @hidden
+   * Error that can be thrown from IExternalAppCommandsService.processActionCommand
+   *
+   * @internal
+   * Limited to Microsoft-internal use
+   *
+   * @beta
+   */
+  export interface ActionCommandError {
+    errorCode: string;
+    message?: string;
+  }
+
+  /**
+   * @internal
+   * Limited to Microsoft-internal use
+   * @hidden
+   * This API will delegate an ActionCommand request to the extensibility provider to send the request to the backend service of the application with provided:
```

Comment:
with provided Id? With provided parameters? The sentence seemed incomplete. 
Also, should we say request is delegated to 'host' following the same pattern from externalAppCardActions?

Reply:
Thanks for helping double check with the comments, I will rephase it to make sure it makes more sense.